jdaviz/app.py

```diff
@@ -1975,8 +2068,27 @@ def set_data_visibility(self, viewer_reference, data_label, visible=True, replac
                 viewer.on_limits_change()  # Trigger compass redraw
 
     def vue_data_item_remove(self, event):
+
+        data = self.data_collection[event['item_name']]
+        self._reparent_subsets(data)
+
         self.data_collection.remove(self.data_collection[event['item_name']])
 
+        if len(self.data_collection) > 1 and len(self.data_collection.external_links) == 0:
+            if self.config == "imviz":
+                link_type = self._jdaviz_helper.plugins["Links Control"].link_type.selected.lower()
+                self._jdaviz_helper.link_data(link_type=link_type, error_on_fail=True)
+                # Hack to restore responsiveness to imviz layers
+                for viewer_ref in self.get_viewer_reference_names():
+                    viewer = self.get_viewer(viewer_ref)
+                    loaded_layers = [layer.layer.label for layer in viewer.layers if
+                                     "Subset" not in layer.layer.label]
+                    self.remove_data_from_viewer(viewer_ref, loaded_layers[-1])
+                    self.add_data_to_viewer(viewer_ref, loaded_layers[-1])
```

Comment:
I expected this to lose layer plot options... but it doesn't seem that's the case?

Reply:
Hm, perhaps the plot option variables don't get deleted on data removal, they're just hidden once the data is gone. That is surprising.

Reply:
So much surprise here worries me. Are we accidentally exploiting a bug that upstream might fix in the future, breaking this code? Or can we ignore?